scripts/helper/functions.sh

```diff
@@ -62,6 +62,28 @@ get_kafka_cluster_id_from_container()
   return 0
 }
 
+build_connect_image()
+{
+  echo
+  echo "Building custom Docker image with Connect version ${CONFLUENT_DOCKER_TAG} and connector version ${CONNECTOR_VERSION}"
+  
+  if [[ "${CONNECTOR_VERSION}" =~ "SNAPSHOT" ]]; then
+    DOCKERFILE="${DIR}/../Dockerfile-local"
+  else
+    DOCKERFILE=$"{DIR}/../Dockerfile-confluenthub"
+  fi
+  echo "docker build --build-arg CP_VERSION=${CONFLUENT_DOCKER_TAG} --build-arg CONNECTOR_VERSION=${CONNECTOR_VERSION} -t localbuild/connect:${CONFLUENT_DOCKER_TAG}-${CONNECTOR_VERSION} -f $DOCKERFILE ."
+  docker build --build-arg CP_VERSION=${CONFLUENT_DOCKER_TAG} --build-arg CONNECTOR_VERSION=${CONNECTOR_VERSION} -t localbuild/connect:${CONFLUENT_DOCKER_TAG}-${CONNECTOR_VERSION} -f $DOCKERFILE . || {
+    echo "ERROR: Docker image build failed. Please troubleshoot and try again. For troubleshooting instructions see https://docs.confluent.io/current/tutorials/cp-demo/docs/index.html#troubleshooting"
+    exit 1
+  }
+  
+  # Copy the updated kafka.connect.truststore.jks back to the host
```

Comment:
@javabrett this PR is fully ready for review (in fact I have done one pass and @awalther28 is working on another pass).  However, we did discuss in Slack that ideally the following steps should be optimized away.  It is written the way it is because of behavior in my environment whereby the changes done during Docker build did not persist on the host (`/tmp` worked, but working on `/etc/kafka/secrets/` did not) -- I was unsuccessful in making this work.  (See Dockerfile for details of steps). Did you want to take another look at this to be able to remove the hack below, or should I commit as is?

Reply:
I think this is OK to commit as-is.  Would you be able to create a quick "can it be improved" JIRA and assign it to me?  I can then look at it a little later for further optimisation.

Reply:
👍 thanks @javabrett .  I filed DEVX-2208 to track this (and assigned to you)